pkg/controller/storagecluster/placement.go

```diff
@@ -33,13 +33,6 @@ func getPlacement(sc *ocsv1.StorageCluster, component string) rookv1.Placement {
 		}
 	}
 
-	topologyMap := sc.Status.NodeTopologies
-	if topologyMap != nil && (component == "mon" || component == "mds") {
-		topologyKey := determineFailureDomain(sc)
-		topologyKey, _ = topologyMap.GetKeyValues(topologyKey)
-		podAffinityTerms := placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
-		podAffinityTerms[0].PodAffinityTerm.TopologyKey = topologyKey
```

Comment:
While this looks like code that does nothing (I thought the same thing when I saw it)
It actually modifies the antiaffinity of `placement` before returning it.
```
placement.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution[0].PodAffinityTerm.TopologyKey = topologyKey
```

Reply:
Thanks, I verified it. Here is the playground link for anyone interested https://play.golang.org/p/MEnYgrnX_jr

Reply:
To elaborate more on previous comment: When you copy a struct in golang, you make a shallow copy. Only those members of the struct that are of the basic data type get copied by value and the rest get copied by reference. On line 40 above, a slice that is a member of the inner struct of the placement is being copied to podAffinityTerms and therefore is copied by reference.

Reply:
Thanks! Yes, that is correct.